azurerm/internal/services/datafactory/tests/data_factory_integration_runtime_self_hosted_test.go

```diff
@@ -0,0 +1,277 @@
+    "commandToExecute" = "powershell -ExecutionPolicy Unrestricted -File gatewayInstall.ps1 ${azurerm_data_factory_integration_runtime_self_hosted.host.auth_key_1} && timeout /t 120"
+  })
+}
+
+resource "azurerm_resource_group" "host" {
+  name     = "acctesthostRG-df-%d"
+  location = "%s"
+}
+
+resource "azurerm_data_factory" "host" {
+  name                = "acctestdfirshh%d"
+  location            = azurerm_resource_group.host.location
+  resource_group_name = azurerm_resource_group.host.name
+}
+
+resource "azurerm_data_factory_integration_runtime_self_hosted" "host" {
+  name                = "acctestirshh%d"
+  data_factory_name   = azurerm_data_factory.host.name
+  resource_group_name = azurerm_resource_group.host.name
+}
+
+resource "azurerm_resource_group" "target" {
+  name     = "acctesttargetRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_role_assignment" "target" {
+  scope                = azurerm_data_factory.host.id
+  role_definition_name = "Contributor"
+  principal_id         = azurerm_data_factory.target.identity[0].principal_id
+}
+
+resource "azurerm_data_factory" "target" {
+  name                = "acctestdfirsht%d"
+  location            = azurerm_resource_group.target.location
+  resource_group_name = azurerm_resource_group.target.name
+
+  identity {
+    type = "SystemAssigned"
+  }
+}
+
+resource "azurerm_data_factory_integration_runtime_self_hosted" "target" {
+  name                = "acctestirsht%d"
+  data_factory_name   = azurerm_data_factory.target.name
+  resource_group_name = azurerm_resource_group.target.name
+
+  rbac_authorization {
+    resource_id = azurerm_data_factory_integration_runtime_self_hosted.host.id
+  }
+
+  depends_on = [azurerm_role_assignment.target, azurerm_virtual_machine_extension.test]
+}
+`, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomInteger, data.RandomInteger, data.RandomString, data.RandomString, data.RandomInteger, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomInteger, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomInteger)
+}
+
+func testCheckAzureRMDataFactoryIntegrationRuntimeSelfHostedExists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := acceptance.AzureProvider.Meta().(*clients.Client).DataFactory.IntegrationRuntimesClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		// Ensure we have enough information in state to look up in API
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		name := rs.Primary.Attributes["name"]
+		factoryName := rs.Primary.Attributes["data_factory_name"]
+		resourceGroup, hasResourceGroup := rs.Primary.Attributes["resource_group_name"]
```

Comment:
Could we use a resource specific parse id function here like can be seen in the analysis package: parse.AnalysisServicesServerID